qiskit/result/result.py

```diff
@@ -305,27 +305,38 @@ def get_counts(self, experiment=None):
         else:
             return dict_list
 
-    def get_statevector(self, experiment=None, decimals=None):
+    def get_statevector(self, experiment=None, decimals=None, label=None):
         """Get the final statevector of an experiment.
 
         Args:
             experiment (str or QuantumCircuit or Schedule or int or None): the index of the
                 experiment, as specified by ``data()``.
             decimals (int): the number of decimals in the statevector.
                 If None, does not round.
+            label (str or None): key name of save statevector used when calling
+                ``save_statevector()``
 
         Returns:
             list[complex]: list of 2^num_qubits complex amplitudes.
 
         Raises:
-            QiskitError: if there is no statevector for the experiment.
+            QiskitError: if there is no statevector for the experiment or if the given
+                statevector label is not in the result.
         """
+
+        if label is None:
+            _label = "statevector"
+        else:
+            _label = label
+
         try:
-            return postprocess.format_statevector(
-                self.data(experiment)["statevector"], decimals=decimals
-            )
+            return postprocess.format_statevector(self.data(experiment)[_label], decimals=decimals)
         except KeyError as ex:
-            raise QiskitError(f'No statevector for experiment "{repr(experiment)}"') from ex
+            if label is None:
+                raise QiskitError(f'No statevector for experiment "{repr(experiment)}"') from ex
+            raise QiskitError(
+                f'No statevector "{label}"' 'for experiment "{repr(experiment)}"'
+            ) from ex
```

Comment:
In line 338, the string has been split in two, and consequently the f-string modifier isn't being applied to the second half, which also needs it.

Also, while an incredibly nitpicky point, I first thought that the two errors being raised were different because of the branching.  Would it perhaps be a shade clearer that they're the same if we do something like
```python
label_text = "" if label is None else (label + " ")
raise QiskitError(
    f'No statevector {label_text}for experiment "{experiment!r}"'
) from ex
```
(i.e. branch on the message, rather than the raise)